internal/app/user_test.go

```diff
@@ -0,0 +1,204 @@
+package app
+
+import (
+	"reflect"
+	"testing"
+	"time"
+
+	"github.com/passwall/passwall-server/internal/config"
+	"github.com/passwall/passwall-server/internal/storage"
+	"github.com/passwall/passwall-server/model"
+	uuid "github.com/satori/go.uuid"
+)
+
+var (
+	mockName     = "passwall-patron"
+	mockEmail    = "patron@passwall.io"
+	mockPassword = "123456789123456789"
+	mockSecret   = "supersecret"
+	now          = time.Now()
+	uuidN        = uuid.NewV4()
+)
+
+func TestGenerateSchema(t *testing.T) {
+	store, err := initDB()
+	if err != nil {
+		t.Errorf("Error in database initialization ! err:  %v", err)
+	}
+	type args struct {
+		s    storage.Store
+		user *model.User
+	}
+	tests := []struct {
+		name    string
+		args    args
+		wantErr bool
+	}{
+		{name: "No error ", args: args{
+			s: store,
+			user: &model.User{
+				ID:               0,
+				UUID:             uuid.NewV4(),
+				Name:             mockName,
+				Email:            mockEmail,
+				MasterPassword:   mockPassword,
+				Secret:           mockSecret,
+				Schema:           "users",
+				Role:             "Member",
+				ConfirmationCode: "12345678",
+				EmailVerifiedAt:  time.Time{},
+			},
+		},
+			wantErr: false,
+		},
+		// todo: populate different cases in test table
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			got, err := GenerateSchema(tt.args.s, tt.args.user)
+			if (err != nil) != tt.wantErr {
+				t.Errorf("GenerateSchema() error = %v, wantErr %v", err, tt.wantErr)
+				return
+			}
+			want, err := store.Users().FindByEmail(mockEmail)
+			if err != nil {
+				t.Errorf("FindByEmail() error %v", err)
+			}
+			got.CreatedAt = got.CreatedAt.UTC()
+			got.UpdatedAt = got.UpdatedAt.UTC()
+			got.EmailVerifiedAt = got.EmailVerifiedAt.UTC()
```

Comment:
FindByEmail() function returns fields which are containing time in UTC timezone, however database records in different timezone. It might seems minor thing to consider however, in case of time oriented operations, it might create some headaches.